src/EditorFeatures/CSharpTest2/Recommendations/WhenKeywordRecommenderTests.cs

```diff
@@ -9,67 +9,233 @@ namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.Recommendations
+            await VerifyAbsenceAsync(AddInsideMethod(@"switch (1) { case (1 + 1 $$: }"));
+            await VerifyAbsenceAsync(AddInsideMethod(@"switch (1) { case (1 + 1 $$ break; }"));
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        public async Task TestForSwitchCase_NotInsideParentheses()
+        {
+            await VerifyAbsenceAsync(AddInsideMethod(@"switch (1) { case (1 + 1 $$) }"));
+            await VerifyAbsenceAsync(AddInsideMethod(@"switch (1) { case (1 + 1 $$): }"));
+            await VerifyAbsenceAsync(AddInsideMethod(@"switch (1) { case (1 + 1 $$) break; }"));
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        public async Task TestForSwitchCase_NotAfterNew()
+        {
+            await VerifyAbsenceAsync(AddInsideMethod(@"switch (1) { case new $$ }"));
+            await VerifyAbsenceAsync(AddInsideMethod(@"switch (1) { case new $$: }"));
+            await VerifyAbsenceAsync(AddInsideMethod(@"switch (1) { case new $$ break; }"));
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        public async Task TestForSwitchCase_NotAfterCase()
+        {
+            await VerifyAbsenceAsync(AddInsideMethod(@"switch (1) { case $$ }"));
+            await VerifyAbsenceAsync(AddInsideMethod(@"switch (1) { case $$: }"));
+            await VerifyAbsenceAsync(AddInsideMethod(@"switch (1) { case $$ break; }"));
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        public async Task TestForSwitchCase_NotAfterDefault()
+        {
+            await VerifyAbsenceAsync(AddInsideMethod(@"switch (1) { default $$ }"));
+            await VerifyAbsenceAsync(AddInsideMethod(@"switch (1) { default $$: }"));
+            await VerifyAbsenceAsync(AddInsideMethod(@"switch (1) { default $$ break; }"));
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        public async Task TestForSwitchCase_NotAfterWhen()
+        {
+            await VerifyAbsenceAsync(AddInsideMethod(@"switch (1) { case 1 when $$ }"));
+            await VerifyAbsenceAsync(AddInsideMethod(@"switch (1) { case 1 when $$: }"));
+            await VerifyAbsenceAsync(AddInsideMethod(@"switch (1) { case 1 when $$ break; }"));
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        public async Task TestForSwitchCase_NotInEmptySwitchStatement()
+        {
+            await VerifyAbsenceAsync(AddInsideMethod(@"switch (1) { $$ }"));
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        public async Task TestForSwitchCase_SemanticCheck_NotAfterPredefinedType()
+        {
+            await VerifyAbsenceAsync(@"
+class SyntaxNode { }
+class Color { }
+class C
+{
+    const Color Color = null;
+    void M() { switch (new object()) { case int $$ } }
```

Comment:
i don't understnd why there is "const Color Colon" in these tests.

Reply:
Because I wanted to have the same environment in this set of tests and just do different things inside the switch statement... but you're right maybe it's not a good idea after all.